sample/sample.go

```diff
@@ -150,10 +50,9 @@ func Sample(src, target string, m int) error {
 
 	var shutdown uint32
 	q := make(chan error)
-	defer close(q)
```

Comment:
Um ponto que eu não consegui resolver e queria tirar dúvida: Esse `defer` do canal estava causando um `panic` nos testes de `sample` com o erro `send on closed channel`, minha suspeita é de que o canal estava sendo fechado antes das go rotinas serem concluídas. Não sei se é algo específico da minha máquina ou um bug de concorrência. De qualquer forma, gostaria de entender melhor o uso desse `shutdown`, e o porque de não usar um `waitGroup` para sincronizar as go rotinas.

Reply:
Isso corre na `main` ou nessa _feature branch_? Se for na `main` a gente pode _debugar_ junto qualquer hora. Mas, enfim… fechar o canal tem que ficar aí. Ia comentar mais pra frente de qualquer maneira. Só quis manter o foco nas coisas novas por enquanto hehehe…

Reply:
Parou de acontecer com as novas alterações que eu subi, então provavelmente eu devo ter introduzido esse bug com as minhas alterações anteriores.